src/vs/base/common/labels.ts

```diff
@@ -36,6 +41,11 @@ export function getPathLabel(resource: URI | string, userHomeProvider: IUserHome
 		resource = URI.file(resource);
 	}
 
+	const labelProvider = UriLabelProviderRegistry.getUriLabelProvider(resource.scheme);
```

Comment:
Not a fan of this mix of "services" and registries. Already today, this function can only work properly in a context in which services are accessible. 

Consider moving the function to a better place (platform?) or use the same duck-type-trick. Then, consider defining one new service for this. The challenge is to create label from an uri. The `IUserHomeProvider` and `IWorkpsaceFolderProvider` are only used for that purpose and we could define one new service interface that is for this, having one provider-interface, not three.